### ISSUE TITLE
Text rendering cuts off whole words in some cases

### ISSUE THREAD

**Issue by Kaiden42** [iced-rs/iced#782]
Hello, I noticed that the text rendering cuts off whole words. It seems that the width of the characters is not calculated correctly. Maybe a rounding mistake?

![text_bug](https://user-images.githubusercontent.com/26521388/111486647-f3502900-8737-11eb-8bbb-d8ea54328d42.png)

Here is a small example, using the latest master, to reproduce this behavior:
```Rust
use iced::{Column, Container, Length, Sandbox, Settings, Text};

fn main() -> iced::Result {
    TextBug::run(Settings::default())
}

struct TextBug {}

#[derive(Debug)]
enum Message {}

impl Sandbox for TextBug {
    type Message = Message;
    fn new() -> Self { TextBug {} }
    fn title(&self) -> String { "TextBug".into() }
    fn update(&mut self, _message: Self::Message) {}
    fn view(&mut self) -> iced::Element<'_, Self::Message> {
        Container::new(
            Column::new()
                .push(Text::new("Toggle Word Wrap"))
                .push(Text::new("Toggle Word wrap"))
                .push(Text::new("Toggle Wörd Wräp"))
                .push(Text::new("Toggle Wird Wrap"))
                .push(Text::new("Toggle Word Wrip"))
                .push(Text::new("Toggle Word? Wrap"))
                .push(Text::new("Toggle Word Wrap?"))
        )
        .width(Length::Fill)
        .height(Length::Fill)
        .center_x()
        .center_y()
        .into()
    }
}
```

**Comment by hecrj** [comment]
I can reproduce the issue on Linux if I manually set a `WINIT_SCALE_FACTOR` of `2.0` and use the default system font.

**Comment by lufte** [comment]
I'm hitting this issue when using an italic font on linux, with everything else set on default values. Any workaround you know of?

**Comment by IohannRabeson** [comment]
Hi, I'm hitting this issue on OSX and Windows as well.
You can see it does not happen with any text. In the following screenshot, the second file name only `file2.txt` is missing. 
<img width="332" alt="Screenshot 2023-01-11 at 18 05 22" src="https://user-images.githubusercontent.com/685196/211937182-f4807357-56ee-4d98-90ba-5029151a7678.png">
I can confirm I get it with this code on OSX (Catalina) and on Windows.

**Comment by GyulyVGC** [comment]
I had this problem when using a font different from the default one. Setting the font used as default font solved the issue in my case. I think this could be due to iced calculating the width basing on the default font width (maybe?).

I'll leave this comment here in case it can be of any help.

**Comment by hecrj** [comment]
Fixed by #1921.
